logger/fields.go

```diff
@@ -0,0 +1,118 @@
+package logger
+
+import (
+	"time"
+
+	"github.com/influxdata/influxdb/pkg/snowflake"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	// TraceIDKey is the logging context key used for identifying unique traces.
+	TraceIDKey = "trace_id"
+
+	// OperationNameKey is the logging context key used for identifying name of an operation.
+	OperationNameKey = "op_name"
+
+	// OperationEventKey is the logging context key used for identifying a notable
+	// event during the course of an operation.
+	OperationEventKey = "op_event"
+
+	// OperationElapsedKey is the logging context key used for identifying time elapsed to finish an operation.
+	OperationElapsedKey = "op_elapsed"
+
+	// DBInstanceKey is the logging context key used for identifying name of the relevant database.
+	DBInstanceKey = "db_instance"
+
+	// DBRetentionKey is the logging context key used for identifying name of the relevant retention policy.
+	DBRetentionKey = "db_rp"
+
+	// DBShardGroupKey is the logging context key used for identifying relevant shard group.
+	DBShardGroupKey = "db_shard_group"
+
+	// DBShardIDKey is the logging context key used for identifying name of the relevant shard group.
+	DBShardIDKey = "db_shard_id"
+
+	// TraceS is a standard prefix for log messages that start a trace.
+	TraceS = "TRACE->S: "
```

Comment:
Also, as a side note, can this be an extra field that's added as log context instead of a prefix? I think adding something like:

```
zap.String("trace", "start")
zap.String("trace", "end")
```

I like those better since they are less intrusive and easier to ignore if you don't like the prefix keys. I would like it if the messages themselves were human readable sentences.

Reply:
More on this, something like:

```
func TraceStart() zapcore.Field {
        return zap.String("trace", "start")
}
```

Reply:
@jsternberg per the PR notes again, `TRACE->S:` and `TRACE->E` were explicitly added so it was easy to identify trace start and end _messages_. Also, per the PR notes, 

> `op_event` currently has two values `evt.start` or `evt.end` indicating the start or end of an operation

Which adds those to the context

Reply:
Removed prefixes